### PR TITLE
[Feat] Remove deprecated APIs

### PR DESCRIPTION
**Pull request by LorDisturbia**
**Why**
Based on #113 

**Changes**

#### TempuraTesting

- Removed deprecated global `test` functions. The `ViewTestCase` API should be used instead.
- Removed `UITestCase` typealias. `ViewTestCase` should be used instead. No other changes should be necessary.

#### Tempura unreleased

- Removed deprecated `dispatch` and `unsafeDispatch` for `NavigationSideEffect` on `AnyStore` and `AnySideEffectContext`. Katana's normal dispatch accepting `Dispatchable`s should be used instead.
- Removed `__unsafeDispatch` and `__unsafeAwaitDispatch` for `NavigationSideEffect` on `ViewController`. The base methods accepting `Dispatchable`s should be used instead.

**Tasks**
* [x] Add relevant tests, if needed
* [x] Add documentation, if needed
* [x] Update README, if needed
* [x] Ensure that the demo project works properly

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 - Make TempuraTesting build on Xcode 12.5 [#111](https://github.com/BendingSpoons/tempura-swift/pull/111)
 - [BREAKING] Use correct screen size on orientation change. This is a breaking change that makes screenSize of Context and VCContext optional. [#110](https://github.com/BendingSpoons/tempura-swift/pull/110)
 - Replace `UIGraphicsBeginImageContextWithOptions` with `UIGraphicsImageRenderer`. [#108](https://github.com/BendingSpoons/tempura-swift/pull/108)
 
+## Tempura unreleased
+
+- Remove deprecated `dispatch` and `unsafeDispatch` for `NavigationSideEffect` on `AnyStore` and `AnySideEffectContext`. Katana's normal dispatch accepting `Dispatchable`s should be used instead. [#114](https://github.com/BendingSpoons/tempura-swift/pull/114)
+- Remove `__unsafeDispatch` and `__unsafeAwaitDispatch` for `NavigationSideEffect` on `ViewController`. The base methods accepting `Dispatchable`s should be used instead. [#114](https://github.com/BendingSpoons/tempura-swift/pull/114)
+- Add Hydra (`>= 2.0.6`) as an explicit dependency. [#113](https://github.com/BendingSpoons/tempura-swift/pull/113)
+
```
Comment (danyf90) [inline CHANGELOG.md:16]: I think we are marking the breaking changes with [BREAKING]

File: Tempura/Navigation/NavigationActions.swift [hunk 2]
```diff
@@ -161,46 +161,6 @@ public struct Hide: NavigationSideEffect {
...
-    return try Hydra.await(self.dispatch(dispatchable))
-  }
-}
-
-extension ViewController {
-  @discardableResult
-  public func __unsafeDispatch<RSE: NavigationSideEffect>(_ dispatchable: RSE) -> Promise<Void> {
-    return self.store.dispatch(dispatchable)
-  }
-
```
Comment (danyf90) [inline Tempura/Navigation/NavigationActions.swift:197]: Is this on purpose? this was not deprecated

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: Tempura/Navigation/NavigationActions.swift (hunk 2 above)
Comment (FEC-bendingspoons) [reply Tempura/Navigation/NavigationActions.swift:197]: I think this should also have been deprecated

**Review by FEC-bendingspoons — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (LorDisturbia) [reply CHANGELOG.md:16]: I'll do it `a valle`

File: Tempura/Navigation/NavigationActions.swift (hunk 2 above)
Comment (LorDisturbia) [reply Tempura/Navigation/NavigationActions.swift:197]: It was completely useless anyway. There's an `__unsafeDispatch` that takes a `Dispatchable`, and `NavigationSideEffect` is a `Dispatchable`. So this overload was actually doing nothing.

**Pull request merged**